network/msgOfInterest.go

```diff
@@ -63,3 +63,19 @@ func MarshallMessageOfInterest(messageTags []protocol.Tag) []byte {
 	topics := Topics{Topic{key: "tags", data: []byte(tags)}}
 	return topics.MarshallTopics()
 }
+
+// MarshallMessageOfInterestMap generates a message of interest message body
+// for the message tags that map to "true" in the map argument.
+func MarshallMessageOfInterestMap(tagmap map[protocol.Tag]bool) []byte {
+	tags := ""
+	for tag, flag := range tagmap {
+		if flag {
+			tags += "," + string(tag)
```

Comment:
Add a comment to protocol/tags.go that `,` cannot be in a tag?

Reply:
Or use a null byte or \x1e for record separator?

Reply:
I think that defining the const separator in `msgOfInterest.go` would do.
Adding a comment in `tags.go` is also a great idea.

Reply:
It's too late to change the encoding (it's already in master and on mainnet); this PR is just adding a second function to produce this encoding from a map rather than a slice (which is just above the added function).  But documenting it is a good idea; will do that.